data/transactions/logic/eval.go

```diff
@@ -885,3 +887,32 @@ func opGlobal(cx *evalContext) {
 	cx.stack = append(cx.stack, sv)
 	cx.nextpc = cx.pc + 2
 }
+
+func opEd25519verify(cx *evalContext){
+	last := len(cx.stack) - 1 // index of PK
+	prev := last -1           // index of signature
+	pprev := prev - 1         // index of data
+
+	var sv crypto.SignatureVerifier
+	if len(cx.stack[last].Bytes) != len(sv) {
+		cx.err = fmt.Errorf("invalid public key")
```

Comment:
errors.New("") for constant strings that don't need formatting

Reply:
fixed.